nemo/collections/asr/parts/submodules/rnnt_loop_labels_computer.py

```diff
@@ -612,12 +728,13 @@ def _before_inner_loop_get_joint_output(self):
         # blank_mask = self.labels == self._blank_index
         self.state.time_indices_current_labels.copy_(self.state.time_indices, non_blocking=True)
         if self.state.alignments is not None:
+            float_dtype = self.state.float_dtype
             self.state.alignments.add_results_masked_no_checks_(
                 active_mask=self.state.active_mask,
                 time_indices=self.state.time_indices_current_labels,
                 logits=logits if self.preserve_alignments else None,
                 labels=self.state.labels if self.preserve_alignments else None,
-                confidence=self._get_confidence_tensor(F.log_softmax(logits, dim=-1))
+                confidence=self._get_confidence_tensor(F.log_softmax(logits, dim=-1)).to(dtype=float_dtype)
```

Comment:
As a point of clarification, why did you need to add this to() call? It doesn't seem to be related to the rest of the changes.

Reply:
I would be happy to avoid this, but without casting, the code will fail with mixed bf16 precision:
- `log_softmax` returns the value of `float32` type in bfloat16 mixed precision (amp)
- alignments storage is initialized with bf16 type, and adding confidence values inside `add_results_masked_no_checks_` will fail

This is the same issue I observed when reviewing Sasha's PR related to TDT confidence #8982 
Since I enabled computing confidence in tests for RNN-T, I caught this bug and fixed it here https://github.com/NVIDIA/NeMo/pull/8972/files#diff-d8ba9ce8e77769e06174cf0d16842d130debb4a289e92fea5296b081f5a4deabR133